### ISSUE TITLE
生成token的概率

### ISSUE THREAD

**Issue by SXxinxiaosong** [Glanvery/LLM-Travel#2]
您好，请教一下    ，下面的代码是否可以计算生成token的概率？

    def cal_prob(self, input_text):
        # 将input_text转换为t5模型的输入格式
        encodings = self.tokenizer(input_text, max_length=512, return_tensors="pt")
        encodings = {k: v.to(self.device) for k, v in encodings.items()}

        # 根据t5模型真实输出，作为target_text
        target_text = self.predict(input_text)

        # 将target_text转换为t5模型的输出格式
        labels = self.tokenizer.encode(target_text, return_tensors="pt", max_length=512, padding="max_length").to(device)

        # 由labels生成decoder_input_ids，需要在前面补0使得长度与labels相同
        decoder_input_ids = torch.cat([torch.zeros_like(labels[:, :1]), labels[:, :-1]], dim=-1).to(device)

        # 计算生成text的概率
        """outputs 包括 loss logits past_key_values encoder_last_hidden_state"""
        outputs = self.model(**encodings, labels=labels, decoder_input_ids=decoder_input_ids)
        logits = outputs.logits
        print(outputs.logits.size())

        return logits

        # 根据t5模型真实输出，作为target_text
        target_text = self.predict(input_text)

        # 将target_text转换为t5模型的输出格式
        labels = self.tokenizer.encode(target_text, return_tensors="pt", max_length=512, padding="max_length").to(device)

        # 由labels生成decoder_input_ids，需要在前面补0使得长度与labels相同
        decoder_input_ids = torch.cat([torch.zeros_like(labels[:, :1]), labels[:, :-1]], dim=-1).to(device)

        # 计算生成text的概率
        """outputs 包括 loss logits past_key_values encoder_last_hidden_state"""
        outputs = self.model(**encodings, labels=labels, decoder_input_ids=decoder_input_ids)
        logits = outputs.logits
        print(outputs.logits.size())

        return logits

**Comment by Glanvery** [comment]
看起来好像没问题

**Comment by SXxinxiaosong** [comment]
谢谢您的回复，还想再请教一下，打印出来的t5的词表大小为32100，但logits输出的大小是32128，请问这是什么问题呢
